internal/backend/httpsvc/metrics.go

```diff
@@ -0,0 +1,78 @@
+package httpsvc
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"go.autokitteh.dev/autokitteh/internal/backend/telemetry"
+	"go.opentelemetry.io/otel/metric"
+)
+
+type apiMetrics struct {
+	counter  metric.Int64Counter
+	duration metric.Int64Histogram
+}
+
+var metrics sync.Map
+
+// metrics    map[string]apiMetrics
+// metricsMux sync.Mutex
+
+func getServiceAPIMetrics(serviceAPI string, t *telemetry.Telemetry) (*apiMetrics, error) {
```

Comment:
i would rename `get` to `acquire` since this has a side effect.

Reply:
done